Source/Scene/Camera.js

```diff
@@ -1010,12 +1012,23 @@ define([
 
             var ray = this.getPickRay(mousePosition, pickGlobeScratchRay);
             rayIntersection = globe.pick(ray, scene, scratchRayIntersection);
-            if (defined(rayIntersection)) {
+
+            if (scene.pickPositionSupported) {
+                depthIntersection = scene.pickPositionWorldCoordinates(mousePosition, scratchDepthIntersection);
```

Comment:
That's a good idea. We could even take it further and cache it for a few frames or more depending on the frame rate, but that's not something I want to explore here.